### ISSUE TITLE
Vehicle behave abnormally in CARLA Town03/04/05

### ISSUE THREAD

**Issue by hwdryan** [guardstrikelab/carla_apollo_bridge#185]
# Describing the problem

Carla-apollo-bridge works fine in CARLA Town01/02. However, when in Town03/04/05, the vehicle will always freak out, the planning, steering and throttle jumps around in dreamview, and the vehicle tries to change to the lanemark. I tried two-way 2 lanes or two-way 4/8 lanes, curved road, and completely straight road. Add showed the same problem.

# Video recording

https://github.com/guardstrikelab/carla_apollo_bridge/assets/44992571/4c1aeba4-72a7-4321-8734-1a1a6fd78ee6

**Comment by hwdryan** [comment]
Hi @PhilWallace , Thank you for replying! I tried re-generate all map data, but the planning still behave abnormally. Could you share your map data here for me to test? 

Also do you know how to change the target speed in apollo-carla? I tried increasing "default_cruise_speed" in module/planning/conf/planning.conf, but it didn't work. Thanks!

**Comment by PhilWallace** [comment]
> Hi @PhilWallace , Thank you for replying! I tried re-generate all map data, but the planning still behave abnormally. Could you share your map data here for me to test?
> 
> Also do you know how to change the target speed in apollo-carla? I tried increasing "default_cruise_speed" in module/planning/conf/planning.conf, but it didn't work. Thanks!

My map files are copied from another project, you can download them and have a try: https://github.com/MaisJamal/Carla_apollo_maps


About the speed problem, it could be due to the controller problem. You can go to "Tasks -> PNC Monitor", turn on PNC monitor, and check whether the actual speed can catch up with the planned speed (there is a window, dynamically showing these data). If not, you may need to adjust the controller.

I have a version of optimized controller, which you could refer to if you are interested: https://github.com/PhilWallace/apollo-8.0.0_control_opt

**Comment by PhilWallace** [comment]
> Hi @PhilWallace , Thank you for replying! I tried re-generate all map data, but the planning still behave abnormally. Could you share your map data here for me to test?
> 
> Also do you know how to change the target speed in apollo-carla? I tried increasing "default_cruise_speed" in module/planning/conf/planning.conf, but it didn't work. Thanks!

Additionally, you can find my demo here: https://www.youtube.com/watch?v=oOJiw21AAGk

You can check your 'speed' window accordingly to see how your controller works.

**Comment by hwdryan** [comment]
Hi @PhilWallace , Thank you again for your suggestions. I tried both your map source and your control config file. As much as I hope to, sadly neither of them solved my problem with abnormal planning and low speed problem. Are you sure your implementation of apollo 8.0 and the carla-apollo bridge from this repo works for you in towns other than 01 and 02?

**Comment by PhilWallace** [comment]
> Hi @PhilWallace , Thank you again for your suggestions. I tried both your map source and your control config file. As much as I hope to, sadly neither of them solved my problem with abnormal planning or low speed problem. Are you sure your implementation of apollo 8.0 and the carla-apollo bridge from this repo works for you in towns other than 01 and 02?

Yes, I tried my implementation in Carla Town 03 and the planning module works fine. As I suggested above, you can check the PNC Monitor to determine what is the specific problem. You can also check the log file of planning and control module. These log files can be very helpful for debugging.
